tests/run/rescue.scala

```diff
@@ -1,18 +1,45 @@
 object lib {
-  inline def (op: => T) rescue[T] (fallback: => T) = try op catch { case _: Throwable => fallback }
-  inline def (op: => T) rescue[T, E <: Throwable] (fallback: E => T) = try op catch { case ex: E => fallback(ex) }
+  inline def (op: => T) rescue[T] (fallback: => T) =
+    try op
+    catch {
+      case _: Throwable => fallback
```

Comment:
Catching Throwable is dangerous since it will catch FatalError which you're not supposed to recover from, alternatively you could catch https://www.scala-lang.org/api/current/scala/util/control/NonFatal$.html.

Reply:
Good suggestion. We also need to ignore non-local return exception `ReturnThrowable`.